Adds JWT token generation methods

vicpatel opened the pull request:
- Generate creates token encoded with user and save it to store
- Updated token generation method usage in  Corredor and Impersonalization

Please take a look.

<!--
Checklist:

1. API changes have been discussed,
2. Source code must be formatted (`go fmt`),
3. Codegen shouldn't produce modified files,
4. Builds pass,
5. Tests pass,
6. Linked to relevant issues

When you are writing pull request title and describing changes, follow
commit message format in the CONTRIBUTING.md in the codebase root.

-->

vicpatel committed:
Adds JWT token generation methods

- Generates token encoded with user and save it to store
- Updated token generation method usage in Corredor and Impersonalization

darh commented on pkg/auth/jwt_token.go:
```diff
@@ -0,0 +1,86 @@
+	}
+)
+
+func (t *token) Generate(ctx context.Context, s tokenStore, i Identifiable) (tokenString string, err error) {
```
How about if we keep the `tokenStore` directly on the `token` struct? This will allow us to call Generate without store and keep it as lightweight as possible + no passing store to the corredor's Connect.

Would that make sense?

vicpatel replied:
Yes

vicpatel committed:
Ensure extra user validation during access token validation

- validate a token exist in store
- when user is suspended, deleted or roles are removed - removes ALL user's tokens

darh committed:
Refactored access token removal logic

darh committed:
Fix access token handling in integration test

darh committed:
Fix access token handling in fed node handshake
